### PR TITLE
3027 add search suggestion when searching by location

### PR DESCRIPTION
**Pull request by SamJamCul**
### Context

https://trello.com/c/V4IecruO/3027-m-add-search-suggestion-when-searching-by-location

### Changes proposed in this pull request

Adds suggested searches for location search. This shows higher radius search counts for searches that have fewer than 3 results. Additional details are in the Trello ticket!

### Guidance to review

image

### CODE REVIEW

**Comment by AgaDufrat** [comment]
[[M] Add search suggestion when searching by location](https://trello.com/c/V4IecruO/3027-m-add-search-suggestion-when-searching-by-location)

**Comment by TobyRet** [comment]
Postcode search within 5 miles, filtering by SEND and part time.
=> Returns a link to 0 courses within 10 miles

http://localhost:3002/results?fulltime=False&hasvacancies=True&l=1&lat=51.4600848&lng=-0.1201408&loc=Acre+Ln%2C+Brixton%2C+London+SW2+5SP%2C+UK&lq=Acre+Lane%2C+London+SW2+5SP%2C+UK&parttime=True&qualifications=QtsOnly%2CPgdePgceWithQts%2COther&rad=5&senCourses=true

zero_courses_link

**Comment by TobyRet** [comment]
Should the copy 'There are no courses matching your search be added in this PR'? E.g. see https://find-postgraduate-teacher-training.education.gov.uk/results?lat=55.8629772&lng=-4.2524605&rad=20&loc=Buchanan%20St,%20Glasgow%20G1%202FF,%20UK&lq=G1%202FF&l=1&sortby=2&fulltime=False&parttime=False&hasvacancies=True&senCourses=False

File: spec/features/suggested_searches_spec.rb [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+          filter_page.find_courses.click
+
+          expect(results_page.suggested_search_links.first.text).to eq("2 courses within 50 miles")
+          expect(results_page.suggested_search_links.second.text).to eq("4 courses across England")
+        end
+      end
+    end
+
+    context " #TODO or move somewhere else" do
```
Comment (defong) [inline spec/features/suggested_searches_spec.rb]: ?

File: spec/features/suggested_searches_spec.rb [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        results_page_request(radius: 5, results_to_return: 0)
+        suggested_search_count_request(radius: 10, results_to_return: 0)
+        suggested_search_count_request(radius: 20, results_to_return: 2)
+      end
+
+      it "doesn't show the link if there are no courses found" do
+        filter_page.load
+        filter_page.by_postcode_town_or_city.click
+        filter_page.location_query.set "SW1P 3BT"
+        filter_page.search_radius.select "10 miles"
```
Comment (defong) [inline spec/features/suggested_searches_spec.rb]: should this not be 5 miles?

**Comment by SamJamCul** [comment]
Bug here, showing both suggested search box and no results ever box

image

http://localhost:3002/results?fulltime=False&hasvacancies=True&l=1&lat=51.4433884&lng=-0.112612&loc=High+Trees%2C+Tulse+Hill%2C+London+SW2+3PS%2C+UK&lq=High+Trees%2C+London+SW2+3PS%2C+UK&parttime=False&qualifications=QtsOnly%2CPgdePgceWithQts%2COther&rad=5&senCourses=true&subjects=53

File: spec/view_objects/results_view_spec.rb [hunk 3]
```diff
@@ -448,6 +462,35 @@
           )
         end
       end
+
+      describe "#suggested_search_visible?" do
+        subject { described_class.new(query_parameters: {}).suggested_search_visible? }
+
+        context "there are 8900 results" do
```
[the remark this answers is not in the document]
Comment (AgaDufrat) [reply spec/view_objects/results_view_spec.rb]: just following the example above https://github.com/DFE-Digital/find-teacher-training/blob/master/spec/view_objects/results_view_spec.rb#L389-L404

**Comment by fofr** [comment]
Nice :100:

**Pull request merged**